Golem/Golem.cs

```diff
@@ -32,21 +34,41 @@ public class Golem : IGolem, IAsyncDisposable
 
         private readonly HttpClient _httpClient;
 
-        private GolemPrice price;
+        private readonly GolemPrice _golemPrice;
+        
         public GolemPrice Price
         {
             get
             {
-                return price;
+                // var price = ProviderConfig.GolemPrice;
+                // _golemPrice.StartPrice = price.StartPrice;
+                // _golemPrice.GpuPerHour = price.GpuPerHour;
+                // _golemPrice.EnvPerHour = price.EnvPerHour;
+                // _golemPrice.NumRequests = price.NumRequests;
+                return _golemPrice;
             }
             set
             {
-                price = value;
+                _golemPrice.StartPrice = value.StartPrice;
+                _golemPrice.GpuPerHour = value.GpuPerHour;
+                _golemPrice.EnvPerHour = value.EnvPerHour;
+                _golemPrice.NumRequests = value.NumRequests;
+
                 OnPropertyChanged();
             }
         }
 
-        public uint NetworkSpeed { get; set; }
+        private uint _networkSpeed;
+        
+        public uint NetworkSpeed
+        { 
+            get =>_networkSpeed;
+            set
+            {
+                _networkSpeed = value;
+                OnPropertyChanged();
```

Comment:
Should we not pass "NetworkSpeed" property name as an arg of `OnPropertyChanged()` ?

Reply:
In `OnPropertyChanged` the default name is the one of the property in which it is used.